hack/prepare-commit-msg-hook.sh

```diff
@@ -0,0 +1,42 @@
+#!/bin/bash
+
+TOP_LEVEL=$(git rev-parse --show-toplevel)
+
+if [[ $TOP_LEVEL == "" ]]; then
+  echo "Error: this script must be called from a git repository"
+  exit 1
+fi
+
+if [[ $1 == "setup" ]]; then
+	if [[ ! -f $TOP_LEVEL/.git/hooks/prepare-commit-msg ]] && [[ ! -h $TOP_LEVEL/.git/hooks/prepare-commit-msg ]]; then
+		ln -s  $TOP_LEVEL/hack/prepare-commit-msg-hook.sh $TOP_LEVEL/.git/hooks/prepare-commit-msg
+	fi
+	exit 0
+fi
+
+COMMIT_MSG_FILE="$1"
+
+# check if commit message is empty
+HAS_SIGNOFF=$(grep -c 'Signed-off-by:' "${COMMIT_MSG_FILE}")
+
+if [[ $HAS_SIGNOFF == "0" ]]; then
+	UNAME=$(git config user.name)
+	UMAIL=$(git config user.email)
+
+	if [[ $UNAME != "" ]] && [[ $UMAIL != "" ]]; then
+
+		COMMIT_COMMENTS=$(sed -n 's/\(^#.*$\)/\1/p' "${COMMIT_MSG_FILE}")
```

Comment:
I think you can skip this comment handling. I did some tests in my IDE and with git cli, the result is always the same.

However, the more I think about this, the more I think we should not automate the sign off, but only check for it and error if it is missing. Reason: at the end of the day signing off has legal impact, you are signing off that you agree to the DCO (https://developercertificate.org/). This shouldn't be automated but a deliberate action IMHO. @beekhof WDYT?

Reply:
makes sense, Will change it to validation only.

Reply:
done